packages/material-ui-unstyled/src/SliderUnstyled/SliderUnstyled.d.ts

```diff
@@ -51,38 +53,43 @@ export interface SliderUnstyledTypeMap<P = {}, D extends React.ElementType = 'sp
      */
     componentsProps?: {
       root?: {
-        as: React.ElementType;
-        styleProps?: Omit<SliderUnstyledTypeMap<P, D>['props'], 'components' | 'componentsProps'>;
+        as?: React.ElementType;
+        styleProps?: Omit<SliderUnstyledTypeMap<P, D>['props'], 'components' | 'componentsProps'> &
+          SliderStylePropsOverrides;
       };
```

Comment:
@mnajdova Are we sure about these types? I'm lost when I look at them. In MUI-X @dtassone has [used any](https://github.com/mui-org/material-ui-x/blob/492c10437c841830958c8bc4e401f9c33fae8962/packages/grid/_modules_/grid/models/gridSlotsComponentsProps.ts#L5)

```suggestion
      root?: any;
```

which might be too loose, but seems still better. 

I mean, for instance, if I want to add an extra `data-` attribute it fails:

```jsx
import * as React from 'react';
import SliderUnstyled from '@material-ui/unstyled/SliderUnstyled';

export default function UnstyledSlider() {
  return (
    <SliderUnstyled componentsProps={{ root: { 'data-foo': 'bar' } }} defaultValue={10} />
  );
}
```

<img width="469" alt="Capture d’écran 2021-06-10 à 22 15 23" src="https://user-images.githubusercontent.com/3165635/121591081-5e5c8380-ca39-11eb-8876-8ea7b58b383d.png">

---

On `styleProps` specifically, I also don't see why we use `SliderUnstyledTypeMap`. As far as I know, the correlation between the props and what's available in this prop is "light'. For instance, in the slider, we have an extra `marked` property:

https://github.com/mui-org/material-ui/blob/d36858d36741bb57facb7f1e91d0d2d4d760b4bb/packages/material-ui-unstyled/src/SliderUnstyled/SliderUnstyled.js#L624

Reply:
I'm in favor of using specific types instead of `any` whenever possible. This way developers get prop validation for free from Typescript (this can be especially important if a component with mandatory props is used in a slot).
The `data-foo` you mentioned is a bit of an edge case as data attributes are not specified in any `HTMLAttributes`. To use them in a type-safe way, module augmentation should be used. However, IMO this should be the only case where developers should resort to module augmentation when providing `componentsProps`.

Reply:
@michaldudak `data-*`, `tabindex`, `aria-*`, etc. we don't accept any other props than `as` and `styleProps`. The issue for me is that we don't have any easy way to know what are types accepted. A developer could use `components` to render something completely different.

Reply:
In this case yes, they'll fail without module augmentation. But we could use what I made in https://github.com/mui-org/material-ui/blob/d587b1eab467739a2a143f0cd49bfc10cc35ba74/packages/material-ui-unstyled/src/SwitchUnstyled/SwitchUnstyled.tsx - when you pass in a component/intrinsic element in `components` prop, the corresponding `componentsProps` expects props for this exact component/element.

Reply:
> when you pass in a component/intrinsic element in components prop, the corresponding componentsProps

@michaldudak This approach seems to have potential 👍 . Should we open a new issue to keep track of this problem?

Reply:
Sure thing: https://github.com/mui-org/material-ui/issues/26810